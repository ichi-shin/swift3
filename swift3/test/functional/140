#!/bin/bash
# Copyright(c)2014 NTT corp. All Rights Reserved.
# DB1-1-8-3_9

. ./common


S3USER=tester

_random | dd of=$tmp.data bs=4k seek=1024 count=1 > /dev/null 2>&1
etag=$(_etag $tmp.data)

_s3_put /bucket

_s3_put /bucket/objectB -T $tmp.data

_random | dd of=$tmp.data bs=4k seek=1024 count=1 > /dev/null 2>&1
_s3_put /bucket/objectA -T $tmp.data

upload_id=$(_s3_post /bucket/object2?uploads | \
    _xpath '/InitiateMultipartUploadResult/UploadId/text()')

_s3_put /bucket/object2?uploadId=${upload_id}\&partNumber=1 \
    -H "x-amz-copy-source: /bucket/objectA" \
    -H "x-amz-copy-source-if-match: $etag" | _filter_curl xml

_sw_get /bucket+segments | _filter_curl | sed -e "s/${upload_id}/UPLOAD_ID/g"
