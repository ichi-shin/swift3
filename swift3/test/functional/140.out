QA output created by 140
> s3curl --id tester -- -X PUT http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X PUT -T /TMP.data http://SWIFT_HOST/bucket/objectB... 200
> s3curl --id tester -- -X PUT -T /TMP.data http://SWIFT_HOST/bucket/objectA... 200
> s3curl --id tester -- -X POST http://SWIFT_HOST/bucket/object2?uploads... 200
> s3curl --id tester -- -X PUT -H x-amz-copy-source: /bucket/objectA -H x-amz-copy-source-if-match: ETAG http://SWIFT_HOST/bucket/object2?uploadId=UPLOAD_ID&partNumber=1... 412
<?xml version='1.0' encoding='UTF-8'?>
<Error>
  <Code>PreconditionFailed</Code>
  <Message>At least one of the preconditions you specified did not hold.</Message>
  <RequestId>TXID</RequestId>
</Error>
> curl -X GET http://SWIFT_HOST/v1/TENANT/bucket+segments... 200
object2/UPLOAD_ID
